transifex/utils/common.py

```diff
@@ -0,0 +1,283 @@
+                    '-c', 'user.email={}'.format(os.environ['GIT_USER_EMAIL']),
+                    'commit', '-m', self.message
+                ],
+                check=True
+            )
+
+    def push(self):
+        """Push branch to the remote."""
+        subprocess.run(['git', 'push', '-u', 'origin', self.branch_name], check=True)
+
+    def open_pr(self):
+        """Create a new PR on GitHub."""
+        if self.pr:
+            raise RuntimeError(
+                'A pull request has already been opened. [{repo}/{pr}]'.format(
+                    repo=self.name,
+                    pr=self.pr.number
+                )
+            )
+
+        self.pr = self.github_repo.create_pull(
+            self.message,
+            'This PR was created by a script.',
+            'master',
+            self.branch_name
+        )
+
+    def merge_pr(self):
+        if not self.pr:
+            raise RuntimeError(
+                'A pull request has not been opened for {repo}:{branch}.'.format(
+                    repo=self.name,
+                    branch=self.branch_name
+                )
+            )
+
+        if self.pr.is_merged():
+            raise RuntimeError(
+                'The pull request [{repo}/{pr}] has already been merged.'.format(
+                    repo=self.name,
+                    pr=self.pr.number
+                )
+            )
+
+        retries = 0
+        while retries <= self.MAX_MERGE_RETRIES:
+            try:
+                self.pr.merge()
+                logger.info('Merged [%s/#%d].', self.name, self.pr.number)
+                break
+            except GithubException as e:
+                # Assumes only one commit is present on the PR.
+                statuses = self.pr.get_commits()[0].get_statuses()
+
+                # Check for any failing Travis builds. If any are found, notify the team
+                # and move on.
+                if any('travis' in s.context and s.state == 'failure' for s in statuses):
+                    logger.info(
+                        'A failing Travis build prevents [%s/#%d] from being merged. Notifying %s.',
+                        self.name, self.pr.number, self.owner
+                    )
+
+                    self.pr.create_issue_comment(
+                        '@{owner} a failed Travis build prevented this PR from being automatically merged.'.format(
+                            owner=self.owner
+                        )
+                    )
+
+                    break
+                else:
+                    logger.info(
+                        'Status checks on [%s/#%d] are pending. This is retry [%d] of [%d].',
+                        self.name, self.pr.number, retries, self.MAX_MERGE_RETRIES
+                    )
+
+                    retries += 1
+
+                    if retries <= self.MAX_MERGE_RETRIES:
+                        # Exponential backoff.
+                        time.sleep(2 ** retries)
```

Comment:
Given that you're in here refactoring, consider changing this to poll every 5 or 10 minutes. I've noticed that the 10 retries we currently make (34 minutes of sleep time) are *just* short of the time required to merge some PRs. If we bump the retry count to 11 we'll end up waiting `(2^(11+1) - 1) / 60 = 68.25` minutes, which is way more than necessary.